### PR TITLE
C3-142 : Issue while create VM with Private Network

### PR DESCRIPTION
**Pull request by avnish30jn**
Jira Id

C3-142

**Problem**

Cannot create vm with networks other than 'VM Network'

**Solution**

Added support for adding the networks specified in the vm struct

**Testing** 

**Unit Testing**

Done. Tried to create vm with VMprivate-network, the vm was assigned with the VMprivate-network, and a local ip is assigned to the vm.

[root@localhost test]# go run vsphereclient.go
Vm created and is powered on
{"IPs":["10.10.24.209","fe80::72cd:ca68:a895:5ae0","fe80::8d39:b744:d6bd:9a0d"]}

### CODE REVIEW

**Comment by sjpotter** [comment]
instead of 

func reconfigureNetworks(vm *VM) ([]types.BaseVirtualDeviceConfigSpec, error)

it should be 

func reconfigureNetworks(vm *VM) (types.BaseVirtualDeviceConfigSpec, error)

and you would append its result, so 

instead of

deviceChangeSpec, err := reconfigureNetworks(vm)

you would have something like

deviceChangeSpecs := []types.BaseVirtualDeviceConfigSpec{}
deviceChangeSpec, err := reconfigureNetworks(vm)
deviceChangeSpecs = append(deviceChangeSpecs, deviceChangeSpec)

File: virtualmachine/vsphere/util.go [hunk 1]
```diff
@@ -400,6 +400,65 @@ func searchTree(vm *VM, mor types.ManagedObjectReference, name string) (*mo.Virt
...
+// createNetworkDeviceSpec : createNetworkDeviceSpec creates the device spec for the network nwMor
+func addNetworkDeviceSpec(nwMor types.ManagedObjectReference, name string) (*types.VirtualDeviceConfigSpec, error) {
+	// create backing object
+	backing := &types.VirtualEthernetCardNetworkBackingInfo{
+		VirtualDeviceDeviceBackingInfo: types.VirtualDeviceDeviceBackingInfo{
+			DeviceName: name,
+		},
+		Network: &nwMor,
+	}
+	// create ehternet card with the backing info
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/util.go]: ethernet

File: virtualmachine/vsphere/util.go [hunk 2]
```diff
@@ -400,6 +400,65 @@ func searchTree(vm *VM, mor types.ManagedObjectReference, name string) (*mo.Virt
...
+	device, err := object.EthernetCardTypes().CreateEthernetCard("e1000", backing)
+	if err != nil {
+		return nil, err
+	}
+	// connect to the newtork when the nic is connected to vm
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/util.go]: network

File: virtualmachine/vsphere/util.go [hunk 3]
```diff
@@ -400,6 +400,65 @@ func searchTree(vm *VM, mor types.ManagedObjectReference, name string) (*mo.Virt
...
+	spec := &types.VirtualDeviceConfigSpec{
+		Operation: types.VirtualDeviceConfigSpecOperationAdd,
+		Device:    device,
+	}
+
+	return spec, nil
+}
+
+// reconfigureNetworks : reconfigureNetworks configures the vm and attach it to the
+// netowrks in the vm structure
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/util.go]: networks

**Review by pankajtandon12 — CHANGES_REQUESTED** [review]
LGTM. Minor comments.
Please test following scenarios as well with changes:
- Use Public Network
- Use both Private & Public Network

**Comment by avnish30jn** [comment]
Tested with below scenarios

Use Public Network
Use both Private & Public Network
Use private network only

**Review by pankajtandon12 — APPROVED** [review]

**Pull request merged**